src/LIBSVM.jl

```diff
@@ -106,7 +106,15 @@ function SVM(smc::SVMModel, y, X, weights, labels, svmtype, kernel)
         unsafe_copyto!(pointer(libsvmweight_label), smc.param.weight_label, nw)
     end
 
-    SVM(svmtype, kernel, weights, size(X,1),
+    if kernel == Kernel.Precomputed || isa(kernel, Function)
+        # in the precomputed case, each data point we want to predict has
```

Comment:
Discaimer: I'm not a maintainer of this package and comments are just suggestions.

Perhaps don't assume the callable object has type `Function`. I can construct objects that can be called like functions but they are not `Function`. So, if it's not one of the built-in types, I'd just assume it's callable. That's more generic. If  a user supplies callable, then they will get an informative error message telling them the object is not callable. 


In line 45 you then do `kernel::K` and make add `K` as a type parameter in line 43.

Reply:
I think it should be working, however if you can help me with constructing a non-`Function` kernel I can add a test case for it.

Reply:
Added a test for non-`Function` callable.